### ISSUE TITLE
sorry!may i know how to use the filter in my c++ program?

### ISSUE THREAD

**Issue by weituotian** [roman380/gdcl.co.uk-mpeg4#20]
i don not really know how to use a custom filter in my computer?
Does it work when other people 's computer don't register the filter?

**Comment by roman380** [comment]
The filters can be used the very same way as other, DirectShow and COM way. Normally they have to be registered, but there are also ways around it: reg-free COM and direct load. I suggest that you google around for this, as I am unlikely to follow up before November 15.

**Comment by weituotian** [comment]
i had search many articles in google but eventually i prefer to read an example program showing how us to use the gdcl filter. i think this project is excellent and the project would be more fantistic by adding a example of usage!

**Comment by weituotian** [comment]
i had search many articles in google but eventually i prefer to read an example program showing how us to use the gdcl filter. i think this project is excellent and the project would be more fantistic by adding a example of usage!

**Comment by roman380** [comment]
A good example is always a good thing, sure. I am however not aware of any clean sample to immediately point to (even though I am sure even some projects on my site can serve the purpose). Maybe I have some free time later and add some code here into repository, but I am not sure when it can happen.

**Comment by weituotian** [comment]
thanks!you are making  a great job!!!

**Comment by amn** [comment]
As far as I know, one can load and test a filter using, for instance, the immensely useful [Graph Studio Next](https://github.com/cplussharp/graph-studio-next) application (to which @roman380 has been and is contributing, to my knowledge) -- just load the filter DLL with"Insert filter from file..." command in the Graph menu, inspect available filters and add them to your graph. In any case, some knowledge into DirectShow architecture is very recommended.

**Comment by LinArcX** [comment]
Hi. I have same issue. I setup a simple win32 application that uses MPEG2-Encoder. It works pretty well. Now i want to replace it with this MPEG4-Encoder. But i don't know how.

I registered both mp4mux.dll and mp4demux.dll via regsvr32.exe on my machine. Should i do some extra works to use it inside my project?

Also which header files should i include in my project?

@amn I tried to do what you've said. when i load `mp4mux.dll` in GraphStudioNext and try to "Scan for CLSID". It gives me this error:

![hpFFO8B2kF](https://user-images.githubusercontent.com/10884422/119622991-fa24a780-be1c-11eb-87cd-6958a4e4ffa0.png)

EDIT-1: When i hit "Search for Error", it says:
> Unknown HRESULT (0x800700C1)

Edit-2: Ok, i figured that out i was opening the wrong dll.(x86 instead of x64) Now i can capture video with this graph:

![2uuO3AWGu2](https://user-images.githubusercontent.com/10884422/119628922-ca789e00-be22-11eb-9f44-ba9a579eb951.png)

Edit: Also i figured that out that i have to include "Module_i.h". And used this code to instantiate the class:

```
hr = CoCreateInstance(CLSID_MuxFilter, NULL, CLSCTX_INPROC_SERVER, IID_PPV_ARGS(&pMpeg4Encoder));
```

But it gives me this linker error:

> 1>Linking...
> 1>main.obj : error LNK2001: unresolved external symbol _CLSID_MuxFilter
> 1>D:\workspace\c++\projects\directshow_graph_tester\Debug\directshow_graph_tester.exe : fatal error LNK1120: 1 unresolved externals
